tests/mock_crust/utils.rs

```diff
@@ -344,13 +352,23 @@ pub fn create_connected_nodes_with_cache_till_split(network: &Network) -> Vec<Te
                     Event::NodeAdded(..) |
                     Event::Connected |
                     Event::Tick => (),
-                    Event::GroupSplit(..) => break 'outer,
+                    Event::GroupSplit(prefix) => {
+                        let _ = split_events.insert(prefix);
+                        if split_events.len() == split_count {
+                            break 'outer;
```

Comment:
I can use `while split_events.len() < split_count` if you prefer, but this won't break us out of the nested loop on line 350.  Not really a big deal - we clear all the rest of the enqueued events just after this anyway, but we'd be calling `split_events.insert(prefix)` more often than required.

Also, it's a set rather than just a counter as we'll receive multiple identical events since we're polling all the nodes here.